bots/sg_modify_scope_by_port.py

```diff
@@ -0,0 +1,187 @@
+            sg.revoke_egress(
+                IpPermissions=[ # only IpPermissions supported with this func !
+                    {
+                        'FromPort': rule[PORT_FROM],
+                        'IpProtocol': rule[PROTOCOL].lower(),
+                        'IpRanges': [
+                            {
+                                'CidrIp': rule[SCOPE]
+                            },
+                        ],
+                        'ToPort': rule[PORT_TO]
+                    },
+                ]
+            )
+            text_output = text_output + ' rule : ' + rule[SCOPE] + ',' + str(rule[PORT_FROM]) + ',' + str(
+                rule[PORT_TO]) + ',' + str(sg_id) + ',' + rule[PROTOCOL].lower() + ' deleted successfully ;'
+
+        except Exception as e:
+            text_output = text_output + f'Error while trying to delete security group. Error: {e}'
+
+        # check if one is existing first ! avoid duplicates and exception
+        found = is_rule_exists_in_sg(sg, rule, direction, scope)
+        if found:
+            return text_output
+
+        try:
+            sg.authorize_egress(
+                IpPermissions=[  # only IpPermissions supported with this func !
+                    {
+                        'FromPort': rule[PORT_FROM],
+                        'IpProtocol': rule[PROTOCOL].lower(),
+                        'IpRanges': [
+                            {
+                                'CidrIp': scope
+                            },
+                        ],
+                        'ToPort': rule[PORT_TO]
+                    },
+                ]
+
+            )
+            text_output = text_output + ' rule : ' + scope + ',' + str(rule[PORT_FROM]) + ',' + str(
+                rule[PORT_TO]) + ',' + str(sg_id) + ',' + rule[PROTOCOL].lower() + ' created successfully ;'
+
+        except Exception as e:
+            text_output = text_output + f'Error while trying to create security group. Error: {e}'
+    else:
+        text_output = text_output + f'Error unknown direction ; \n'
+
+    return text_output
+
+
+def run_action(boto_session, rule, entity, params):
+    text_output = 'Run bot sg_modify_scope_by_port.'
+    sg_id = entity['id']
+    # Param retrieving
+    try:
+        port, change_from_scope, change_to_scope, direction = params
+    except Exception as e:
+        text_output = text_output + 'Params handling error. Please check parameters and try again.\n ' + e + '\n' \
+                      + 'Usage: AUTO: sg_modify_scope_by_port <port> <change_scope_from|*> <change_scope_to> ' \
+                        '<direction> '
+        raise Exception(text_output)
+
+    ec2_resource = boto_session.resource('ec2')
+    sg = ec2_resource.SecurityGroup(sg_id)
+
+    for rule in entity[f'{direction}Rules']:
+        if rule[PORT_FROM] <= int(port) <= rule[PORT_TO] and change_to_scope != rule[SCOPE]:
+            if change_from_scope == rule[SCOPE] or change_from_scope == '*':
```

Comment:
you already know that change_to_scope != rule[SCOPE] and you check change_from_scope == rule[SCOPE]

Reply:
fixed

Reply:
change_to_scope != rule[SCOPE] removed